### ISSUE TITLE
Systemd-boot loader takes very long to start

### ISSUE THREAD

**Issue by RicArch97** [NixOS/nixpkgs#207295]
### Describe the bug
When updating my system about 2 weeks ago, systemd-boot suddenly started up very slow.
I'm not quite sure yet which commit is to blame, but here are the results of `systemd-analyze time`:
```
Startup finished in 11.716s (firmware) + 22.115s (loader) + 7.347s (kernel) + 7.766s (userspace) = 48.945s
graphical.target reached after 7.766s in userspace.
```
22 seconds for loader... I have a powerful system booting from an NVME drive, with a Ryzen 9 5950X CPU and 4000MHz RAM, so these results are quite unexpected.
My thinkpad T470, which is significantly slower, still takes 7 seconds to start loader, but in my experience that's also slower than it used to be.
My bootloader config is simply
```nix
boot.loader = {
  systemd-boot.enable = true;
  systemd-boot.configurationLimit = 5;
  efi.canTouchEfiVariables = true;
};
```

### Steps To Reproduce
Unsure yet. I have done a full wipe of my drive, cleared BIOS CMOS and then reinstalled my system, yet same result.

### Expected behavior
I expect the systemd-boot bootloader to start within like 3 seconds.

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.13, NixOS, 23.05 (Stoat), 23.05.20221218.04f574a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - nixpkgs: `/nix/store/v3h6hhxf4f5l5ix3p70jap046zl2plm1-source`
```

**Comment by winterqt** [comment]
cc @flokli @Kloenk @Mic92

**Comment by flokli** [comment]
Can you check what version of sd-boot you have installed?

**Comment by RicArch97** [comment]
> Can you check what version of sd-boot you have installed?

version 252.1

**Comment by flokli** [comment]
Hm, this could be https://github.com/systemd/systemd-stable/commit/1c9e7fc8f28347e7151baa7163f0d5bb2cdb9611, which is included in 252.3 and upwards.

252.3 should be in staging, and 252.4 comes with https://github.com/NixOS/nixpkgs/pull/207119. Can you check with one of these versions? You should be able to just invoke `bootctl install` from one of these builds, without having to rebuild the world.

**Comment by bb2020** [comment]
Can you force merge staging into master?
252.1 is still preventing Macbooks from booting.

**Comment by flokli** [comment]
This should be part of the next staging-next cycle. cc @vcunat

**Comment by RicArch97** [comment]
Solved with the systemd 252.4 update. Loader start time is now 2.233 seconds.
